californium-core/src/main/java/org/eclipse/californium/core/network/MessageExchangeStore.java

```diff
@@ -181,6 +184,15 @@
 	boolean isEmpty();
 
 	/**
+	 * Checks if there are any exchanges currently being registered in this
+	 * store and dump these exchanges. Intended to be used for more details in
+	 * log when testing.
+	 * 
+	 * @return {@code true} if there no exchanges registered.
+	 */
+	boolean isEmptyWithDump();
```

Comment:
Should we really add this to the interface ? Maybe add this to the Implementation is enough ?

Reply:
I tend to agree with Simon that `isEmptyWithDump` should not be part of the interface. I also do not see why we need a separate method for this at all. Why don't we simply make `InMemoryMessageExchangeStore.isEmpty` log more detailed information if a finer log level is active, e.g. FINEST?

Reply:
:-) Because in the meantime I poll "isEmpty()" and fail with a more relaxed timeout :-).

Though the polling would cause a lot of logging, I introduced the isemptyWithDump for the final check, to see, if it's not empty, what is left.

Please remember:
The eclipse CI build infrastructure doesn't grant a minimum CPU quote and starvation seems to affect some, but not all threads. So the test thread may execute while the sweeper may starve much more. Therefore the relaxed times are required.
   
Currently I just merge this into the "hudson_debug" branch, if I want to analyse a specific failing test. Any other hudson build failures with not empty exchange stores, I simply ignore in the meantime, because with no additional information it's a waste of time.